dashboard/lib/school_info_interstitial_helper.rb

```diff
@@ -24,6 +24,21 @@ def self.show_school_info_interstitial?(user)
     return true
   end
 
+  # Show the school info confirmation dialog when a teacher has either completely
+  # filled out the school info interstitial for a US public, private, or charter school
+  # or confirmed current school over a year ago.
+  def self.show_school_info_confirmation_dialog?(user)
+    return false unless user.teacher?
+
+    school_info = user.school_info
+
+    check_school_type = (school_info.public_school? || school_info.private_school? || school_info.charter_school?) && school_info.completely_filled_out?
+
+    check_last_confirmation_date = user.user_school_infos.find(user.school_info_id).last_confirmation_date.to_datetime < 1.year.ago
```

Comment:
I think this will work if a user doesn't have school info. Do you have a test or something to confirm?

Reply:
Added test.  Thanks!

Reply:
But what about if user.school_info is nil (which may or may not be the same as a school_info having all nil values)?

Reply:
Thanks for adding the test for this!